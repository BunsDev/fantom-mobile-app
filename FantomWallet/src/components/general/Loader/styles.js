import { StyleSheet, Dimensions } from 'react-native';

// Calculate  max height and width of device
const deviceWidth = Dimensions.get('window').width;
const deviceHeight = Dimensions.get('window').height;

export default StyleSheet.create({
  overlayView: {
    flex: 1,
    position: 'absolute',
    left: 0,
    top: 0,
    opacity: 1.0,
    backgroundColor: 'rgba(100, 100, 100, 0)',
    width: deviceWidth,
    height: deviceHeight,
    zIndex: 12200,
  },
  activityIndicator: {
    alignItems: 'center',
    justifyContent: 'center',
    height: 80,
  },
});
