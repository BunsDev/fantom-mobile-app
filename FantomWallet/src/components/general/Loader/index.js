// @flow
// Libraries
import React from 'react';
import { View, Dimensions, ActivityIndicator } from 'react-native';

import styles from './styles';

// Calculate  max height of device
const deviceHeight = Dimensions.get('window').height;


/**
 * @param  {} {loaderStyle - decimal height factor (?)
 * @param  {} loaderColor
 * @param  {Props} isLoading=false}
 */
export default (props: TLoaderTypes) =>
{
  const { loaderStyle, loaderColor, isLoading = false }=props
  const activityIndicatorColor = loaderColor || 'rgba(0,0,0,1)';
  const marginStyle = loaderStyle ? deviceHeight * loaderStyle - 40 : deviceHeight * 0.5 - 40;
  if (isLoading && isLoading === false) {
    return null;
  }
  return (
    <View style={styles.overlayView}>
      <ActivityIndicator
        animating
        style={[styles.activityIndicator, { marginTop: marginStyle }]}
        size="large"
        color={activityIndicatorColor}
      />
    </View>
  );
};
