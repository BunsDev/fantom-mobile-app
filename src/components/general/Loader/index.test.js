import React from 'react';
import { shallow } from 'enzyme';

import Loader from './index';

test('renders correctly', () => {
  const wrapper = shallow(<Loader loaderStyle={{ padding: 2 }} loaderColor="red" isLoading />);
  expect(wrapper).toMatchSnapshot();
});
