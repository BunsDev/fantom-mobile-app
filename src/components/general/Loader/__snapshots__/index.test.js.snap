// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(100, 100, 100, 0)",
      "flex": 1,
      "height": 1334,
      "left": 0,
      "opacity": 1,
      "position": "absolute",
      "top": 0,
      "width": 750,
      "zIndex": 12200,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="red"
    hidesWhenStopped={true}
    size="large"
    style={
      Array [
        Object {
          "alignItems": "center",
          "height": 80,
          "justifyContent": "center",
        },
        Object {
          "marginTop": NaN,
        },
      ]
    }
  />
</View>
`;
